amd_openvx/openvx/ago/ago_haf_gpu_special_filters.cpp

```diff
@@ -1540,8 +1541,13 @@ int HafGpu_ScaleGaussianHalf(AgoNode * node)
 			"  L0.s01 = vload2(0, (__local uint *)&lbuf_ptr[%d]);\n" // LMemStride * 4
 			"  sum.s0 += (float)(L0.s0 & 0xffff); sum.s1 += (float)(L0.s0 >> 16); sum.s2 += (float)(L0.s1 & 0xffff); sum.s3 += (float)(L0.s1 >> 16);\n"
 			"  sum = sum * (float4)0.00390625f;\n"
-			"  if (valid) {;\n"
-			"    *(__global uint *)p0_buf = amd_pack(sum);\n"
+			"  if (valid) {\n"
+            "    if (validRow) {"
+			"        *(__global uint *)p0_buf = amd_pack(sum);\n"
+            "    }\n"
+            "    else {\n"
+            "        *(__global uint *)p0_buf = 0;\n" // pad the first and last row
```

Comment:
@hansely Padding the first and last row with zero inside the kernel is not a good idea as it has performance implications by adding extra checks for the validRow. The better fix is to initialize the output buffer with zero after allocating it. Then, inside the kernel here, fill the output buffer for the appropriate gy (i.e., 1 < gy < (height - 1)).